vault/cluster.go

```diff
@@ -321,6 +321,11 @@ func (c *Core) startClusterListener(ctx context.Context) error {
 		// If we listened on port 0, record the port the OS gave us.
 		c.clusterAddr.Store(fmt.Sprintf("https://%s", c.getClusterListener().Addr()))
```

Comment:
Could this result in raft using a bad listener address if the `cluster_addr` is, say `127.0.0.1:0` (a valid address with an unspecified port), and the address in the listener is `0.0.0.0:8200` since we'd be overwriting `c.clusterAddr` with one that raft is unhappy about?